contents/security/passwordStorage.md

```diff
@@ -0,0 +1,150 @@
+  head: head.md
+  siteNav: mainNav.md
+  pageNav: 3
+</frontmatter>
+
+{{ navbar | safe }}
+
+<div class="website-content">
+
+# Introduction to Password Storage
+
+**Author: [Jeremy Choo](https://github.com/ChooJeremy)** <br>
+Reviewers: [Amrut Prabhu](https://github.com/amrut-prabhu), [Marvin Chin](https://github.com/marvinchin), [Tan Zhen Yong](https://github.com/Xenonym), [Wang Junming](https://github.com/junming403)
+
+## Overview
+
+Many software applications use a username and password combination as user account credentials for authentication. Obviously, it is not a good idea for the software to store these credentials as <trigger for="pop:plaintext">plain text</trigger> because if someone else were to gain access to them either lawfully (e.g., an employee who has access to the data) or unlawfully (e.g., someone hacking into the data storage), that person can use those credentials directly to impersonate the account owner. This article explains some techniques that can be used to store user credentials more securely:
+
+- Encryption
+- Hashing
+- Salting
+
+<popover id="pop:plaintext" title="_Plaintext_ refers to unencrypted information" placement="top">
+  <div slot="content">
+It is a cryptography term generally referring to text before encyption or after decrypting it. Another term for it is cleartext.
+  </div>
+</popover>
+
+## What is encryption?
+
+Encryption refers to the process of changing a message such that it becomes essentially random text. The opposite process is called _decryption_, which converts the text back into the original message. In order to do this, it takes in an <trigger for="pop:encrypt-key">encryption key</trigger>, which is then re-used in the decryption process. 
+
+Additionally, even a small change in the message will result in a completely different <trigger for="pop:ciphertext">ciphertext</trigger>. This means that even very similar passwords that vary only by capitalization or a letter would still result in different ciphertext, so attackers can't compare the hash to try and figure out which users have similar passwords.
+
+<popover id="pop:ciphertext" title="_Ciphertext_ refers to encrypted information" placement="top">
+  <div slot="content">
+It is a cryptography term generally referring to data after encrypting it.
+  </div>
+</popover>
+
+For example, the following python code encrypts and decrypts the message "EncryptionIsFun!" with <trigger for="pop:aes">AES</trigger>:
+```python
+from Crypto.Cipher import AES
+
+AESCipher = AES.new("Here's a AES Key", AES.MODE_CBC, "Here's a AES IV.")
+message = "EncryptionIsFun!"
+ciphertext = AESCipher.encrypt(message)
+
+AESCipher2 = AES.new("Here's a AES Key", AES.MODE_CBC, "Here's a AES IV.")
+plaintext = AESCipher2.decrypt(ciphertext)
+
+print(plaintext))
+```
+
+<popover id="pop:aes" title="_AES_ stands for Advanced Encryption Standard" placement="top">
+  <div slot="content">
+It is one of many possible encryption algorithms.
+  </div>
+</popover>
```

Comment:
When I asked for an example, i didn't mean python code. Rather, explain the concept using a concrete example. You can find some inspiration here https://www.howtogeek.com/howto/33949/htg-explains-what-is-encryption-and-how-does-it-work/